tpgtools/templates/resource.go.tmpl

```diff
@@ -76,7 +76,11 @@ func Resource{{$.PathType}}() *schema.Resource {
 {{- end }}
 			Delete: schema.DefaultTimeout({{$.DeleteTimeoutMinutes}} * time.Minute),
 		},
-
+{{- if $.CustomCode.ValidateRawResourceConfigFuncs }}
```

Comment:
tpgtools uses a separate resource object entirely than MMv1, so this should mirror the tpgtools/resource.go code you've added like
```suggestion
{{- if $.ValidationFunction }}
```

Then the string should just be output directly, there is no `CustomTemplate` function.

Do you know of a DCL resource needing this functionality already? DCL is on "keeping the lights on" mode and we generally don't need to add new features to tpgtools resources unless it will be immediately important. We have long term goals to migrate off of DCL/tpgtools entirely and are slowly doing so

Reply:
So it may be easier to just remove the tpgtools changes unless you think it will be necessary for a known resource.

Reply:
Compute disk is not DCL for what it's worth, here are the list of resources generated by DCL:
https://github.com/hashicorp/terraform-provider-google-beta/blob/main/google-beta/provider/provider_dcl_resources.go#L37 @BBBmau

Reply:
Thanks I was looking in the wrong place for the DCL resources. Removing it from tpgtools then since no DCL resources require this. https://github.com/GoogleCloudPlatform/magic-modules/pull/13048#discussion_r1954937728